generator/tests/Google/Service/Generator/ResourceTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Google\Service\Generator;
+
+class ResourceTest extends \PHPUnit\Framework\TestCase {
+  private $emptyNode = [];
+
+  function testGetClassName() {
+    $names = ['bidders',
```

Comment:
Do you have other test cases for these methods? If so, consider using the dataprovider.